src/core/smack-setup.c

```diff
@@ -107,12 +109,12 @@ static int write_access2_rules(const char *srcdir) {
                                 continue;
                         }
 
-                        if (write(isempty(acc2) ? load2_fd : change_fd, buf, strlen(buf)) < 0) {
-                                if (r == 0)
-                                        r = -errno;
-                                log_error_errno(errno, "%s/%s: failed to write '%s' to '%s': %m",
+                        q = RET_NERRNO(write(isempty(acc2) ? load2_fd : change_fd, buf, strlen(buf)));
+                        if (q < 0) {
+                                log_error_errno(q, "%s/%s: failed to write '%s' to '%s': %m",
                                                 srcdir, entry->d_name,
                                                 buf, isempty(acc2) ? "/sys/fs/smackfs/load2" : "/sys/fs/smackfs/change-rule");
+                                RET_GATHER(r, q);
```

Comment:
Maybe embedded logger in `RET_GATHER()`?
```c
RET_GATHER(r, log_error_errno(q, "..."));
```

Reply:
I didn't do that because it'd create very deep nesting…

Reply:
But we do that at many places.

Reply:
OK. Not a strong opinion.